wordpress-mobile/WordPress-iOS
# Hiding dashboard cards: add Personalize Home Tab screen

Related issue: #20296

Previous PRs (dependencies):

1. #20365

This PR introduces a new "Personalize Home Tab" button to the Home tab (Jetpack-only). It opens [a new screen](https://jetpackmobile.wordpress.com/2023/03/14/hiding-dashboard-cards-i2-ios/) that allows toggling visibility for _most_ of the cards on the dashboard.

> Not every card is made personalizable yet (see `DashboardCard.personalizableCards` that controls the list). The remaining cards require special attention and will be covered separately.

## Testing Instructions

*Prerequisites*
- "Personalize Home Tab" feature flag is enabled

**Test Case 1**
- Select site A and scroll to the bottom of the Home tab
- Tap "Personalize Home Tab"
- Use the switches to hide one of the cards
- Verify that the card got hidden on the Home tab
- Switch to site B and open the Home tab
- Verify that the previously hidden card is still visible (preferences are saved on a per-site basis)

_Testing Notes_
- Test on iOS 14 and other version to make sure the design is consistent across platforms and everything works as expected
- Test on iPad
- Try different font sizes (dynamic type), themes (light/dark), RTL languages, VoiceOver

## Screenshots

| Add button to Home tab      | Add Personalize screen |
| ----------- | ----------- |
|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 19 00 22](https://user-images.githubusercontent.com/1567433/226484580-45db9da1-3b4d-4671-8d8c-a8a7947fbd23.png)| ![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 19 00 08](https://user-images.githubusercontent.com/1567433/226484591-f0389ad5-5a6e-4091-8d93-0d114de01182.png) |

| Dynamic Type      | Dynamic Type |
| ----------- | ----------- |
| ![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 19 00 22](https://user-images.githubusercontent.com/1567433/226484855-76205485-057e-432f-8e93-cfcfca23a2b5.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 19 00 14](https://user-images.githubusercontent.com/1567433/226484813-79203330-cadd-4aa2-9789-65eb8957ca09.png) |

| Dark Mode      | Dark Mode |
| ----------- | ----------- |
|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 20 15 15](https://user-images.githubusercontent.com/1567433/226492286-35ee3dce-5843-48cb-86a8-3750bceee641.png) |![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 20 15 19](https://user-images.githubusercontent.com/1567433/226492309-20cc2615-e134-42db-bc6d-8de7494a3bca.png) 

| RTL      | RTL |
| ----------- | ----------- |
| ![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 18 59 20](https://user-images.githubusercontent.com/1567433/226484911-7a33729a-ea95-4ced-8e71-281ed025c63f.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2023-03-20 at 18 59 26](https://user-images.githubusercontent.com/1567433/226484921-983176bf-13d7-4e0a-bf6a-a8824bd277bb.png) |

|iPad|
| ----------- | 
| ![Screenshot 2023-03-27 at 3 50 13 PM](https://user-images.githubusercontent.com/1567433/228051768-0fad803c-789d-4815-a838-c6a75b92f5e7.png) |

## Regression Notes
1. Potential unintended areas of impact: **n/a** (it's a new screen, so I don't expect it to affect any of the existing features)
3. What I did to test those areas of impact (or what existing automated tests I relied on): **n/a**
4. What automated tests I added (or what prevented me from doing so): I added a few unit tests for the new screen

PR submission checklist:
